scan/src/utils/ID.re

```diff
@@ -0,0 +1,26 @@
+type t =
```

Comment:
Let's make custom types for **each** ID type.
```
Module DataSource {
    type t = ID(int)
}

...

type t = 
   | DID(DataSource.t)
   | OID(...)
   | RID(...)
   ...
```

We will be a lot safer in our lives.